File: code/innobyte/group-50/Queue.java
// Queue data structure task implemented in java
// Author: Okem Emmanuel Friday
// Reg No: 20/CSC/263

public class Queue<T> {
    private static final int DEFAULT_CAPACITY = 10; // Default capacity of the queue
    private T[] elements; // Array to store elements of the queue
    private int size; // Current number of elements in the queue
    private int front; // Index of the front element
    private int rear; // Index of the rear element

    // Constructor to initialize the queue with default capacity
    public Queue() {
        this(DEFAULT_CAPACITY);
    }

    // Constructor to initialize the queue with a specified capacity
    public Queue(int capacity) {
        elements = (T[]) new Object[capacity]; // Create an array of specified capacity to store elements
        size = 0; // Initialize the size to zero as the queue is empty initially
        front = 0; // Initialize the front index
        rear = -1; // Initialize the rear index to -1 as there are no elements in the queue
    }

}

File: code/innobyte/group-53/Queue.py
# Queue Data Structure task implementation in Python
# Author: Oko Emmanuel Adoshi
# Reg No: 20/CSC/106

class Queue:
    def __init__(self):
        """Initialize an empty queue."""
        self.items = []

    def is_empty(self):
        """Check if the queue is empty."""
        return len(self.items) == 0
